docs/input/docs/upgrading/1.x-to-2.x.md

```diff
@@ -0,0 +1,211 @@
+
+### Update bootstrapper to include --bootstrap command
+
+Cake.Recipe now includes the use of the .Net Global Tool Module.  As such, it will be necessary to first bootstrap this module, before executing the build.  The can be done as follows:
+
+```powershell
+Write-Host "Bootstrapping Cake"
+dotnet cake $SCRIPT_NAME --bootstrap
+if ($LASTEXITCODE -ne 0) { exit $LASTEXITCODE }
+
+Write-Host "Running Build"
+dotnet cake $SCRIPT_NAME @args
+if ($LASTEXITCODE -ne 0) { exit $LASTEXITCODE }
+```
+
+### Rename AppVeyor default target
+
+Due to the fact that Cake.Recipe is no longer directly coupled to AppVeyor, the entry target name has changed.  Instead of calling:
+
+```yml
+build_script:
+  - ps: .\build.ps1 -Target AppVeyor
+```
+
+you should call:
+
+```yml
+build_script:
+  - ps: .\build.ps1 --target=CI
+```
+
+### Update GitVersion.yml file with the required changes for using new version of GitVersion
+
+Cake.Recipe has switched from using 3.6.5 of GitVersion, to 5.0.1 of the .Net Framework version, and 5.3.7 of the .Net Global Tool version.  There have been a number of breaking changes in how GitVersion configuration works, specifically around how branches are configured.  If you have done any branch configuration in the GitVersion.yml file, you may have to update to the [latest standards](https://gitversion.net/docs/configuration#branch-configuration).
+
+### Ensure AppVeyor build image is using latest
+
+In order to execute correctly on AppVeyor, you will need to use the Visual Studio 2019 build image.  Change the image definition from:
+
+```yml
+image: Visual Studio 2017
+```
+
+to:
+
+```yml
+image: Visual Studio 2019
+```
+
+### Ensure that Environment.SetVariableNames() is called
+
+Previously, the call to [Environment.SetVariableNames](../fundamentals/set-variable-names) was not mandatory, and could be omitted from the recipe.cake file.  This is no longer the case, and it must be included.
+
+## Suggested Changes
+
+### Use embedded icon for NuGet package
+
+This guidance comes from the NuGet team directly.  Going forward, you should look to include an icon within the nupkg file.
+
+### Add CakeContrib.Guidelines Nuget Package
+
+To help with the above suggestion, and likely others given time, we are recommending that you add the [CakeContrib.Guidelines](https://www.nuget.org/packages/CakeContrib.Guidelines/) NuGet package to your project.  This will inspect your project, and make sure that you are following the guidelines from the Cake-Contrib Organisation for your addin.
+
+### Remove usage of nuspec file(s) for NuGet packages
+
+Due to the fact that Cake.Recipe has now enabled fully deterministic builds, as well as turning on of SourceLink and embedding pdb files into the generated nupkg file, it is simply easier to run `dotnet pack` rather than attempting to do this manually.  To faciliate this, you will need to add the following parameter:
```

Comment:
```suggestion
Due to the fact that Cake.Recipe has now enabled fully deterministic builds, as well as turning on of SourceLink and embedding pdb files into the generated nupkg file, it is simply easier to run `dotnet pack` rather than attempting to do this manually.  To facilitate this, you will need to add the following parameter:
```

Reply:
Additionally:
```suggestion
Due to the fact that Cake.Recipe has now enabled fully deterministic builds, as well as turning on of SourceLink and embedding pdb files into the generated nupkg file, and it is simply easier to run `dotnet pack` rather than attempting to do this manually.  To faciliate this, you will need to add the following parameter:
```

Reply:
I have made a few additional changes to this part.